### ISSUE TITLE
UMN backup failure

### ISSUE THREAD

**Issue by kuroi-hayabusa** [mvysny/aedict#849]
I messed up my Notepad. I could not find the db file in the advised folder (…/sk.baka.aedict3/files/umn/dbfiles/db). I searched all over all the android storage folders, trying from the PC or directly from the phone. Then I moved back to Stored Type “File-based” and, as advised, I lost data.

Now I am trying to get familiar with the new UMN system back up. I still do not see where the files are stored: the advised folder only contains the dictionary files.

I uninstalled and re-installed back the app with the same result.

Aedict version is 3.50.7 and the phone is oneplus 5.

**Comment by kuroi-hayabusa** [comment]
Hi, Actually I see some files which seems to be the dictionary ones, but I am not interested in those. It is my notepad file what I can not find. (the "show hidden files" Android option is enabled).

I am not sure if I got this: is there any way I can back up my date without synchronizing with Aedict Online?
I mean something like copying the db file somewhere and get it back when needed?
Regards

**Comment by mvysny** [comment]
With UMN, everything is packed in one database file; I don't remember exactly how it's called, wild guess would be `userdata.db`. if you're interested, I can check that up for you.

You're right, back up is impossible. However you can set up sync to your own server if you'd like. Google for UMN server if you're interested. If you wish, I can prepare a manual on how to do that.

**Comment by kuroi-hayabusa** [comment]
Yes, could you please check the location and name of the file. I am curious, I modify the notepad as a test, but I cannot find any updated folder/file in the phone?
Also guidelines on sync'ing it the would be much appreciated.

**Comment by mvysny** [comment]
Actually the location of the database file varies; for me it's `/data/user/0/sk.baka.aedict3/files/UMN/dbfiles/` and there's a file simply named `db`. 
It's not visible because it's private to aedict so that rogue apps can't read it. You can however use adb or Android studio to read the file. You can read more info here:  https://groups.google.com/forum/?nomobile=true#!topic/aedict-users/DyWrbwCv9m8

**Comment by mvysny** [comment]
Syncing to multiple servers may be tricky. Let me think more about it.

**Comment by mvysny** [comment]
Related feature request: https://gitlab.com/mvysny/umn/issues/3
